lib/ansible/modules/windows/win_package.ps1

```diff
@@ -319,7 +319,12 @@ if ($state -eq "absent") {
                 $cleanup_artifacts += $local_path
             } elseif ($program_metadata.location_type -eq [LocationType]::Empty -and $program_metadata.msi -ne $true) {
                 # TODO validate the uninstall_string to see if there are extra args in there
-                $local_path = $program_metadata.uninstall_string
+                $parsed_uninstall_string = @(ParseCommandLine $program_metadata.uninstall_string)
```

Comment:
`ParseCommandLine` isn't a cmdlet I'm aware off which means the code isn't actually doing anything. Can't you just do

```
$local_path = $program_metadata.uninstall_string
if ($local_path.StartsWith('"') -and $local_path.EndsWith('"')) {
    $local_path = $local_path.Substring(1, $local_path.Length - 2)
}
```

Reply:
`ParseCommandLine` was in Ansible.ModuleUtils.CommandUtil.psm1 and thrown away with 190d1ed7f1cd6be4746e937906d571f0abe4a53d
https://github.com/ansible/ansible/blob/691ff4b9e61a019633e6da139539d4464b5557e5/lib/ansible/module_utils/powershell/Ansible.ModuleUtils.CommandUtil.psm1#L182-L197



What about values for uninstall_string like 
`"C:\Program Files\myprog\uninst.exe" Folder="C:\Documents and Settings\All Users\Application Data\myprog"`

Reply:
Ahh ok I see what you were trying to do. Unfortunately that wouldn't have worked as you are calling the .NET method but you haven't specified the class it was in. You would have technically had to do `[Ansible.CommandUtil]::ParseCommandLine($program_metadata.uninstall_string)` but the namespace has been changed from that commit you mentioned so it won't work anymore.

The code was changed to better align with the newer C# module util structure we have recently introduced. At the time of the commit I couldn't find it being used anywhere else and we decided to make the change.

Ultimately I think what we should be doing is to not process the uninstall registry value at all and just pass that directly into `Run-Command`. That way any comments or arguments that are stored by the app are preserved and we can just append our arguments onto the end of the string if they have been set by the user. This would mean some changes further down to not call `Argv-ToString` but it should still be possible in this case.